ludwig/config_validation/checks.py

```diff
@@ -234,6 +235,33 @@ def check_validation_metric_exists(config: "ModelConfig") -> None:  # noqa: F821
         )
 
 
+@register_config_check
+def check_retrieval_config(config: "ModelConfig") -> None:
+    """Checks that the retrieval config is valid."""
+    for input_feature in config.input_features:
+        if input_feature.type == TEXT:
+            if input_feature.preprocessing.prompt.task is not None:
+
+                _check_k_retrieval_config(input_feature.preprocessing.prompt.retrieval)
+                _check_model_name_retrieval_config(input_feature.preprocessing.prompt.retrieval)
+
+
+def _check_k_retrieval_config(retrieval_config: "RetrievalConfig") -> None:
+    """Checks that k is greater than zero if retrieval type is not None."""
+    if retrieval_config.type is None and retrieval_config.k != 0:
```

Comment:
We can clean this up in the future by having a dynamically loaded schema based on the selection of the `type` param. That way we can auto-enforce that `k` isn't set if the type is empty, etc. This can be a follow-up PR, but I would leave a TODO for now.

Reply:
Done